### ISSUE TITLE
ValueError: With n_samples=0, test_size=0.2 and train_size=0.8, the resulting train set will be empty. Adjust any of the aforementioned parameters.

### ISSUE THREAD

**Issue by Ying-Jung** [WarrenGreen/srcnn#6]
ValueError: With n_samples=0, test_size=0.2 and train_size=0.8, the resulting train set will be empty. Adjust any of the aforementioned parameters.

Can someone tell me how can I fix it? Thanks for everything .

**Comment by sagar1899** [comment]
in preprocess.py file generate_samples function doen not generate 400*400 dimension images.

give proper path

**Comment by Sinnovate** [comment]
> in preprocess.py file generate_samples function doen not generate 400*400 dimension images.
> 
> give proper path

Hi @sagar1899,
cud u able to run this in ur system.....if yes can u pls share the code & the places u modified fr the sme.......

**Comment by sagar1899** [comment]
> > in preprocess.py file generate_samples function doen not generate 400*400 dimension images.
> > give proper path
> 
> Hi @sagar1899, cud u able to run this in ur system.....if yes can u pls share the code & the places u modified fr the sme.......

yes i'm able to run this code in my system.

in preprocess.py file first it will generate smaller size multiple images in "raw" directory. then it will generate train and test data in "train" and "test" directory. then it will generate data in "train_labels" and "test_labels". 
provide proper paths in functions in preprocess.py.

**Comment by emmajreid** [comment]
I believe the problem is that in the util.py file, it only adds to y if the y_path is None.

**Comment by sagar1899** [comment]
> I believe the problem is that in the util.py file, it only adds to y if the y_path is None.

Yes you have to change code "y path is not none in util.py"

**Comment by Sadrim-71** [comment]
> > > in preprocess.py file generate_samples function doen not generate 400*400 dimension images.
> > > give proper path
> > 
> > 
> > Hi @sagar1899, cud u able to run this in ur system.....if yes can u pls share the code & the places u modified fr the sme.......
> 
> yes i'm able to run this code in my system.
> 
> in preprocess.py file first it will generate smaller size multiple images in "raw" directory. then it will generate train and test data in "train" and "test" directory. then it will generate data in "train_labels" and "test_labels". provide proper paths in functions in preprocess.py.

Thanks Sagar. How much does this software need RAM for training? Because I faced memory errors (having about 20Gb). 
did this also happen to you? 
I used the sample image provided by the author. I also reduced the extent to 25% of the original image, but still facing memory error. 
I would be grateful to have your comments.

**Comment by sagar1899** [comment]
> > > > in preprocess.py file generate_samples function doen not generate 400*400 dimension images.
> > > > give proper path
> > > 
> > > 
> > > Hi @sagar1899, cud u able to run this in ur system.....if yes can u pls share the code & the places u modified fr the sme.......
> > 
> > 
> > yes i'm able to run this code in my system.
> > in preprocess.py file first it will generate smaller size multiple images in "raw" directory. then it will generate train and test data in "train" and "test" directory. then it will generate data in "train_labels" and "test_labels". provide proper paths in functions in preprocess.py.
> 
> Thanks Sagar. How much does this software need RAM for training? Because I faced memory errors (having about 20Gb). did this also happen to you? I used the sample image provided by the author. I also reduced the extent to 25% of the original image, but still facing memory error. I would be grateful to have your comments.

Yes I already face this problem, I have 8 gb ram in my pc and i can train upto 800-1000 images in training dataset. So make small dataset and train on it.

**Comment by Sadrim-71** [comment]
> > > > > in preprocess.py file generate_samples function doen not generate 400*400 dimension images.
> > > > > give proper path
> > > > 
> > > > 
> > > > Hi @sagar1899, cud u able to run this in ur system.....if yes can u pls share the code & the places u modified fr the sme.......
> > > 
> > > 
> > > yes i'm able to run this code in my system.
> > > in preprocess.py file first it will generate smaller size multiple images in "raw" directory. then it will generate train and test data in "train" and "test" directory. then it will generate data in "train_labels" and "test_labels". provide proper paths in functions in preprocess.py.
> > 
> > 
> > Thanks Sagar. How much does this software need RAM for training? Because I faced memory errors (having about 20Gb). did this also happen to you? I used the sample image provided by the author. I also reduced the extent to 25% of the original image, but still facing memory error. I would be grateful to have your comments.
> 
> Yes I already face this problem, I have 8 gb ram in my pc and i can train upto 800-1000 images in training dataset. So make small dataset and train on it.

thanks. you are right. it worked.

**Comment by VJdroid** [comment]
> > > > > > in preprocess.py file generate_samples function doen not generate 400*400 dimension images.
> > > > > > give proper path
> > > > > 
> > > > > 
> > > > > Hi @sagar1899, cud u able to run this in ur system.....if yes can u pls share the code & the places u modified fr the sme.......
> > > > 
> > > > 
> > > > yes i'm able to run this code in my system.
> > > > in preprocess.py file first it will generate smaller size multiple images in "raw" directory. then it will generate train and test data in "train" and "test" directory. then it will generate data in "train_labels" and "test_labels". provide proper paths in functions in preprocess.py.
> > > 
> > > 
> > > Thanks Sagar. How much does this software need RAM for training? Because I faced memory errors (having about 20Gb). did this also happen to you? I used the sample image provided by the author. I also reduced the extent to 25% of the original image, but still facing memory error. I would be grateful to have your comments.
> > 
> > 
> > Yes I already face this problem, I have 8 gb ram in my pc and i can train upto 800-1000 images in training dataset. So make small dataset and train on it.
> 
> thanks. you are right. it worked.

Hi @Sadrim-71 can please share the full code, i got multiple errors related to path. Iam unable to clear the error.
I would be thankful to have answer from you.

Thank you.

**Comment by Sadrim-71** [comment]
[mycrnn_v2.zip](https://github.com/WarrenGreen/srcnn/files/10298841/mycrnn_v2.zip)
Hello @VJdroid 
take a look at it. i hope it helps. 
best wishes

**Comment by sagar1899** [comment]
Hi,
Find the attached file that contains code.

Thanks,
Sagar Khimani

**Comment by PuieMarian** [comment]
I have the same error, trying to perform super resolution on satellite images using a convolutional neural network (CNN).
Can someone help please?
